Assignment 2.Rmd

```diff
@@ -85,15 +89,49 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 ## Part II
 
-In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+#In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+(a1<- count(D1, stid))
+(a2<- count(D1, tid))
+names(a2) <- c("stid","n")
+a3 <- rbind(a1,a2)
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender,vertex.size = a3$n+15)
+```
 
 
 
 ## Part III
 
-Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+#Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+#Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+library(readr)
+DC1 <- read_csv("hudk4050-classes.csv")
+library(dplyr)
+#install.packages("tidyr")
+library(tidyr)
+library(igraph)
```

Comment:
You don't need to load the package `igraph` again since you loaded it in Part I.